test/Mono.Linker.Tests.Cases/DataFlow/PropertyDataFlow.cs

```diff
@@ -265,6 +268,36 @@ public void TestPropertyWithExistingAttributes ()
 				[param: DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicConstructors)]
 				set { PropertyWithExistingAttributes_Field = value; }
 			}
+
+			public void TestPropertyWithIndexerWithMatchingAnnotations ([DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicConstructors)] Type myType)
```

Comment:
This needs RecognizedReflectionAccessPattern attribute - otherwise the test doesn't actually validate anything - it might generate warnings, but the test will not fail as is.